client/StaticThreadTracing.java

```diff
@@ -0,0 +1,219 @@
+package grabl.tracing.client;
+
+import grabl.tracing.client.GrablTracing.Analysis;
+import grabl.tracing.client.GrablTracing.Trace;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class StaticThreadTracing {
```

Comment:
This class appears to be quite low-level and contains a lot of know-hows. Can we describe the behaviour of this class to help the reader understand why it's needed, what it does, how it does it, and also document in any special behaviours (constructors or methods) in a javadoc comment of that code?